conformance/utils/kubernetes/helper.go

```diff
@@ -2,18 +2,72 @@ package kubernetes
 
 import (
 	"context"
+	"fmt"
+	"net"
 	"strings"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/require"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2ekubectl "k8s.io/kubernetes/test/e2e/framework/kubectl"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"sigs.k8s.io/network-policy-api/conformance/utils/config"
 )
 
+// PokeServer is a utility function that checks if the connection from the provided clientPod in clientNamespace towards the targetHost:targetPort
+// using the provided protocol can be established or not and returns the result based on if the expectation is shouldConnect or !shouldConnect
+func PokeServer(t *testing.T, clientNamespace, clientPod, protocol, targetHost string, targetPort int32, timeout time.Duration, shouldConnect bool) bool {
+	t.Helper()
+	cmd := []string{"exec", clientPod, "--"} // command is to be run inside a pod
+	timeoutArg := fmt.Sprintf("--timeout=%v", timeout)
+	protocolArg := fmt.Sprintf("--protocol=%s", protocol)
+	ipPortArg := net.JoinHostPort(targetHost, fmt.Sprintf("%d", targetPort))
+
+	// we leverage the dial connect from agnhost, that is already supporting multiple protocols
+	connectCommand := strings.Split(fmt.Sprintf("/agnhost connect %s %s %s",
+		timeoutArg,
+		protocolArg,
+		ipPortArg), " ")
+
+	cmd = append(cmd, connectCommand...)
+	var res string
+	var err error
+	res, err = e2ekubectl.RunKubectl(clientNamespace, cmd...)
+	// TODO(tssurya): Improve the error matching to be more specific (https://pkg.go.dev/k8s.io/kubernetes/test/images/agnhost#readme-connect)
```

Comment:
Let's eventually match this into an ENUM like what's done in the core k8s utils here -> https://github.com/kubernetes/kubernetes/blob/ce05a4f7fcc234dd3132a4e40e6afdf0b2ebc94e/test/e2e/framework/network/utils.go#L936

Reply:
ack let me open an issue for this..: https://github.com/kubernetes-sigs/network-policy-api/issues/108